### PR TITLE
feat: Monitor barcode scanning performance

### PR DESCRIPTION
**Pull request by g123k**
Monitor how long the barcode scanner took #4622

### CODE REVIEW

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/analytics/AnalyticsTrackingEvent.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+import android.app.Activity
+import java.util.concurrent.TimeUnit
+
+fun Activity.startTrackEvent(event : AnalyticsEvent): AnalyticsTrackingEvent {
+    return AnalyticsTrackingEvent(event)
+}
+
+data class AnalyticsTrackingEvent(
+    val startDate: Long,
```
Comment (VaiTon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/analytics/AnalyticsTrackingEvent.kt]:
```suggestion
    val startDate: Long = System.currentTimeMillis(),
```
and remove the secondary constructor

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/analytics/AnalyticsTrackingEvent.kt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    val event: AnalyticsEvent
+) {
+
+    constructor(event: AnalyticsEvent) : this(System.currentTimeMillis(), event)
+
+    fun computeDurationInSeconds() =
+        TimeUnit.MILLISECONDS.toSeconds(
+            System.currentTimeMillis() - startDate
+        ).toFloat()
```
Comment (VaiTon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/analytics/AnalyticsTrackingEvent.kt]: If i remember correctly we can use something like 
`(System.currentTimeMillis() - startDate).milliseconds.toSeconds()`

Comment (g123k) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/analytics/AnalyticsTrackingEvent.kt]: Didn't know that method, thanks!

**Review by VaiTon — APPROVED** [review]

**Comment by VaiTon** [comment]
Idk why detekt is not working. Merged develop to see if it's an issue

**Comment by teolemon** [comment]
good to merge ?

**Review by VaiTon — APPROVED** [review]

**Pull request merged**